src/http/server.cr

```diff
@@ -186,6 +182,41 @@ class HTTP::Server
     server.local_address
   end
 
+  # Creates a `OpenSSL::SSL::Server` and adds it as a socket.
+  #
+  # The SSL server wraps a `TCPServer` listenting on `host:port`.
+  #
+  # ```
+  # server = HTTP::Server.new { }
+  # context = OpenSSL::SSL::Context::Server.new
+  # context.certificate_chain = "openssl.crt"
+  # context.private_key = "openssl.key"
+  # server.bind_ssl "127.0.0.1", 8080, context
+  # ```
+  def bind_ssl(host : String, port : Int32, context : OpenSSL::SSL::Context::Server = OpenSSL::SSL::Context::Server.new, reuse_port : Bool = false) : Socket::IPAddress
```

Comment:
Shouldn't `bind_ssl` methods be wrapped between `{% unless flag?(:without_openssl) %}`?
Otherwise there is a reference to a type that is not declared.